lib/wasmer-types/src/values.rs

```diff
@@ -204,3 +222,78 @@ impl<T> From<ExternRef> for Value<T> {
 //         Self::FuncRef(val)
 //     }
 // }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_value_i32_from_u32() {
+        let input: u32 = 0x00000000;
+        let value = Value::<()>::from(input);
+        let mut out: i128 = 0;
+        unsafe {
+            value.write_value_to(&mut out as *mut i128);
+        }
+        assert_eq!(out, 0x00000000);
```

Comment:
It looks like my other comment got lost somehow, I intended to comment on this and say I think we can just do:

```Rust
        let value = Value::<()>::from(0u32);
        assert_eq!(value, Value::I32(0));
```

here.  I think this tests all the same things and is easier to read in my opinion!

Reply:
That is true and easy for `u32`s in the signed integer range. But what about

```rust
        let value = Value::<()>::from(0xffffffffu32);
        assert_eq!(value, Value::I32(/* what goes here now? */));
```

The idea was to ensure the same bytes go in and out, avoiding decimal representation.

Reply:
Got a bit confused by the GitHub UI and responded in a top level comment! https://github.com/wasmerio/wasmer/pull/1624#issuecomment-706408935